NetKet/Machine/rbm_spin.hpp

```diff
@@ -70,7 +80,22 @@ class RbmSpin : public AbstractMachine<T> {
   }
 
   void Init() {
-    W_.resize(nv_, nh_);
+    if (usewavelets_) {
+      W_wavelets_.resize(nw_)
+
+      Wavelets_.resize(nv_, nw_)
+      // initialize wavelets using gabor wavelets
+      for (int i = 0; i < nv_; i++) {
+        for (int j = 0; j < nw_; j++) {
+            // NOTE: there I need to know if I'm in 1D or 2D
+            Wavelets_(i, j) = 0.;
```

Comment:
I think that the solution is to create a new wavelet class that takes a `Hypercube` object as a constructor. 
`Hypercube basically contains all information about the the number of sites, geometry, periodic boundary conditions etc.

Reply:
Ok I found it https://github.com/netket/netket/blob/v2.0/NetKet/Graph/hypercube.hpp . Is there an example of an existing class that would look like `Wavelets`I could inspire myself from ?